persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -101,24 +83,30 @@ class AbstractPersistentEntityRegistry(system: ActorSystem, injector: Injector)
 
     val cluster = Cluster(system)
 
-    cluster.registerOnMemberUp {
-      if (role.forall(cluster.selfRoles.contains)) {
-        val entityProps = PersistentEntityActor.props(
-          persistenceIdPrefix = entityTypeName, Optional.empty(), entityFactory, snapshotAfter, passivateAfterIdleTimeout,
-          journalPluginId, snapshotPluginId
-        )
-        sharding.start(prependName(entityTypeName), entityProps, shardingSettings, extractEntityId, extractShardId)
-      } else {
-        // not required role, start in proxy mode
-        sharding.startProxy(prependName(entityTypeName), role, extractEntityId, extractShardId)
+    def start(): Unit = {
+      cluster.registerOnMemberUp {
+        if (config.runEntitiesOnRole.forall(cluster.selfRoles.contains)) {
+          val entityProps = PersistentEntityActor.props(
+            persistenceIdPrefix = entityTypeName, Optional.empty(), entityFactory, config.snapshotAfter.asJava,
+            config.passivateAfterIdleTimeout, journalPluginId, snapshotPluginId
+          )
+          sharding.start(prependName(entityTypeName), entityProps, shardingSettings, extractEntityId, extractShardId)
+        } else {
+          // not required role, start in proxy mode
+          sharding.startProxy(prependName(entityTypeName), config.runEntitiesOnRole, extractEntityId, extractShardId)
+        }
       }
     }
+
+    if (config.delayRegistration) {
+      cluster.registerOnMemberUp(start())
+    } else start()
```

Comment:
`registerOnMemberUp` is used both in `start()` and outside.

Reply:
That explains why the tests are still failing, thanks for the catch!